shm/src/MapAlternativeVersion.cpp

```diff
@@ -0,0 +1,47 @@
+#include "MapAlternativeVersion.hpp"
+
+#include <algorithm>
+#include <random>
+#include <string>
+
+constexpr uint16_t numberOfIslands = 10;
+constexpr uint16_t axisX = 100;
+constexpr uint16_t axisY = 100;
+
+Map::Map() {
+    std::vector<Island> islands_(numberOfIslands);
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> genPositionX(0, axisX);
+    std::uniform_int_distribution<> genPositionY(0, axisY);
+    std::vector<std::pair<uint16_t, uint16_t>> positionsBefore;
```

Comment:
Instead of another field (which will be used only in c-tor ) you can use existing Islands :)